### PR TITLE
Added options to QuerySummary and linked the QueryDuration to it

### PR DESCRIPTION
**Pull request by huppeje**
https://coveord.atlassian.net/browse/JSUI-2112

JSUI-2112: Added options to QuerySummary and linked the QueryDuration to it

### CODE REVIEW

File: src/ui/QuerySummary/QuerySummary.ts [hunk 1]
```diff
@@ -19,8 +19,10 @@ import { IQuery } from '../../rest/Query';
 import { IQueryResults } from '../../rest/QueryResults';
 
 export interface IQuerySummaryOptions {
+  enableResultsSummary?: boolean;
+  enableNoResultsForSearch?: boolean;
+  enableCancelLastAction?: boolean;
   enableSearchTips?: boolean;
-  onlyDisplaySearchTips?: boolean;
```
Comment (olamothe) [inline src/ui/QuerySummary/QuerySummary.ts]: We cannot remove options like this, as this is a breaking change in the component API.

The same component configuration with the same option should yield the same resulting behaviour after the change. We can add new option, we can mark some older one as "deprecated", but we cannot remove option, or modify how they behave in a non major/breaking version.

We have a mechanism in place to rename option if needed.

File: test/ui/QuerySummaryTest.ts [hunk 2]
```diff
@@ -135,25 +135,25 @@ export function QuerySummaryTest() {
 
       it('onlyDisplaySearchTips allow to not render the results range', () => {
         test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
-          onlyDisplaySearchTips: false
+          // onlyDisplaySearchTips: false
```
Comment (olamothe) [inline test/ui/QuerySummaryTest.ts]: The same UT should still be working after the changes. It's generally a good sign that the changes we did were non-breaking :)

File: src/ui/QuerySummary/QuerySummary.ts (hunk 1 above)
Comment (huppeje) [reply src/ui/QuerySummary/QuerySummary.ts]: It totally makes sense! I will make the adjustments!

File: test/ui/QuerySummaryTest.ts (hunk 2 above)
Comment (huppeje) [reply test/ui/QuerySummaryTest.ts]: Good point :)

**Pull request closed without merging**